client/src/components/SingleSelectField/SingleSelectField.js

```diff
@@ -98,14 +51,33 @@ class SingleSelectField extends Component {
   }
 
   render() {
-    let field = null;
-    if (this.props.readOnly) {
-      field = this.getReadonlyField();
-    } else {
-      field = this.getSelectField();
+    // .slice() to copy the array, because we could modify it with an empty item
+    const options = (this.props.source)
+      ? this.props.source.slice()
+      : [];
+
+    if (this.props.data.hasEmptyDefault && !options.find((item) => !item.value)) {
+      options.unshift({
+        value: '',
+        title: this.props.data.emptyString,
+        disabled: false,
+      });
     }
 
-    return field;
+    return (
+      <Input type="select" {...this.getInputProps()}>
```

Comment:
@maxime-rainville Is there built-in support for the `readOnly` prop in this Reactstrap component? Or does this change remove the readOnly mode?

Reply:
The `else` statement in `getInputProps` above disables the select field when the field is readonly.

I was getting weird errors with the current split logic, so I figured simplifying the code was a nice side effect of this fix.

Reply:
Oh gotcha, that was tricky to spot!